hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/OmTableInsightTask.java

```diff
@@ -0,0 +1,373 @@
+                  (k, size) ->
+                      size > omKeyInfo.getReplicatedSize() ?
+                          size - omKeyInfo.getReplicatedSize() : 0L);
+            }
+          }
+          break;
+
+        default:
+          LOG.trace("Skipping DB update event : Table: {}, Action: {}",
+              omdbUpdateEvent.getTable(), omdbUpdateEvent.getAction());
+        }
+      } catch (Exception e) {
+        LOG.error(
+            "Unexpected exception while processing the table {}, Action: {}",
+            omdbUpdateEvent.getTable(), omdbUpdateEvent.getAction(), e);
+        return new ImmutablePair<>(getTaskName(), false);
+      }
+    }
+
+    // Write count data to DB if the sizeCountMap is not empty
+    if (!objectCountMap.isEmpty()) {
+      writeDataToDB(objectCountMap);
+    }
+
+    // Write unreplicated size data to DB if the sizeCountMap is not empty
+    if (!unreplicatedSizeCountMap.isEmpty()) {
+      writeDataToDB(unreplicatedSizeCountMap);
+    }
+
+    // Write replicated size data to DB if the sizeCountMap is not empty
+    if (!replicatedSizeCountMap.isEmpty()) {
+      writeDataToDB(replicatedSizeCountMap);
+    }
+
+    LOG.info("Completed a 'process' run of OmTableInsightTask.");
+    return new ImmutablePair<>(getTaskName(), true);
+  }
+
+
+  private void writeDataToDB(Map<String, Long> dataMap) {
+    List<GlobalStats> insertGlobalStats = new ArrayList<>();
+    List<GlobalStats> updateGlobalStats = new ArrayList<>();
+
+    for (Entry<String, Long> entry : dataMap.entrySet()) {
+      Timestamp now =
+          using(sqlConfiguration).fetchValue(select(currentTimestamp()));
+      GlobalStats record = globalStatsDao.fetchOneByKey(entry.getKey());
+      GlobalStats newRecord
+          = new GlobalStats(entry.getKey(), entry.getValue(), now);
+
+      // Insert a new record for key if it does not exist
+      if (record == null) {
+        insertGlobalStats.add(newRecord);
+      } else {
+        updateGlobalStats.add(newRecord);
+      }
+    }
+
+    globalStatsDao.insert(insertGlobalStats);
+    globalStatsDao.update(updateGlobalStats);
+  }
+
+  private HashMap<String, Long> initializeCountMap() {
+    Collection<String> tables = getTaskTables();
+    HashMap<String, Long> objectCountMap = new HashMap<>(tables.size());
+    for (String tableName : tables) {
+      String key = getTableCountKeyFromTable(tableName);
+      objectCountMap.put(key, getValueForKey(key));
+    }
+    return objectCountMap;
+  }
+
+  private HashMap<String, Long> initializeReplicatedSizeMap() {
+    Collection<String> tables = getTablesToCalculateSize();
+    HashMap<String, Long> sizeCountMap = new HashMap<>(tables.size());
+    for (String tableName : tables) {
+      String key = getReplicatedSizeKeyFromTable(tableName);
+      sizeCountMap.put(key, getValueForKey(key));
+    }
+    return sizeCountMap;
+  }
+
+  private HashMap<String, Long> initializeUnReplicatedSizeMap() {
```

Comment:
I think above 2 initialisation methods can be combined to one to avoid 2 for loops.

Reply:
I have combined both the methods into one!